pilot/pkg/security/authn/policy_applier.go

```diff
@@ -36,3 +36,10 @@ type PolicyApplier interface {
 	// It may return nil, if no authentication is needed.
 	AuthNFilter(proxyType model.NodeType) *http_conn.HttpFilter
 }
+
+// MtlsDecider is the interace to decide wether a particular workload at a specific port
+// is able to accept mTLS or not.
+type MtlsDecider interface {
```

Comment:
Why do we need a new interface? Would it be simpler to add this function to the existing applier interface?

Reply:
applier interface is implemented by alpha as well. we have diff use cases here. blending small interface is perfectly fine.

https://www.integralist.co.uk/posts/go-interfaces/#keep-interfaces-small